app/importers/file_importers/star_math_importer.rb

```diff
@@ -1,18 +1,34 @@
 class StarMathImporter < Struct.new :school_scope, :client, :log, :progress_bar
 
   def import
-    return unless remote_file_name
+    return unless zip_file_name.present? && remote_file_name.present?
 
-    @data = CsvDownloader.new(
-      log: log, remote_file_name: remote_file_name, client: client, transformer: data_transformer
-    ).get_data
+    log.write("\nDownloading ZIP file #{zip_file_name}...")
 
-    @data.each.each_with_index do |row, index|
-      import_row(row) if filter.include?(row)
-      ProgressBar.new(log, remote_file_name, @data.size, index + 1).print if progress_bar
+    downloaded_zip = client.download_file(zip_file_name)
+
+    Zip::File.open(downloaded_zip) do |zipfile|
+      zipfile.each do |file|
+        if file.name == remote_file_name
```

Comment:
This is an ugly/hacky way to find the right CSV within the ZIP file, though it worked. A cleaner way would be to use the `#read` API on Zip::File: http://www.rubydoc.info/github/rubyzip/rubyzip/Zip/File#read-instance_method